ocs_ci/ocs/jenkins.py

```diff
@@ -0,0 +1,265 @@
+        self.create_namespace()
+        self.build_completed = []
+
+    def create_ocs_jenkins_template(self):
+        """
+        Create jenkins template
+
+        """
+        try:
+            ocs_jenkins_template = templating.load_yaml(constants.JENKINS_TEMPLATE)
+            self.ocs_jenkins_template = OCS(**ocs_jenkins_template)
+            self.ocs_jenkins_template.create()
+        except(CommandFailed, CalledProcessError) as cf:
+            log.error('Failed to create jenkins template')
+            raise cf
+
+    def create_namespace(self):
+        """
+        create namespace for Jenkins
+        """
+        if self.ocp is None:
+            self.ocp = OCP(namespace=constants.JENKINS_NAMESPACE)
+            self.ocp.new_project(constants.JENKINS_NAMESPACE)
+
+    def setup_jenkins_build_config(self):
+        """
+        Setup jenkins build config
+
+        """
+        try:
+            jenkins_build_config = templating.load_yaml(constants.JENKINS_BUILDCONFIG_YAML)
+            self.jenkins_build_config = OCS(**jenkins_build_config)
+            self.jenkins_build_config.create()
+        except(CommandFailed, CalledProcessError) as cf:
+            log.error('Failed to create Jenkins build config')
+            raise cf
+
+    def wait_for_jenkins_deploy_status(self, status, timeout=None):
+        """
+        Wait for jenkins deploy pods status to reach running/completed
+
+        Args:
+            status (str): status to reach Running or Completed
+            timeout (int): Time in seconds to wait
+
+        """
+        timeout = timeout if timeout else 600
+        # Wait for jenkins-deploy pods to reached and running
+        log.info(f"Waiting for jenkins-deploy pods to be reach {status} state")
+        jenkins_deploy_pods = self.get_jenkins_deploy_pods()
+        for jenkins_deploy_pod in jenkins_deploy_pods:
+            try:
+                wait_for_resource_state(
+                    resource=jenkins_deploy_pod, state=status, timeout=timeout
+                )
+            except ResourceWrongStatusException:
+                cmd = f'logs {jenkins_deploy_pod.name}'
+                output = self.ocp.exec_oc_cmd(command=cmd, out_yaml_format=False)
+                error_msg = f'{jenkins_deploy_pod.name} did not reach to {status} state after {timeout} sec\n{output}'
+                log.error(error_msg)
+                raise UnexpectedBehaviour(error_msg)
+
+    def wait_for_build_status(self, status, timeout=None):
+        """
+        Wait for build status to reach running/completed
+
+        Args:
+            status (str): status to reach Running or Completed
+            timeout (int): Time in seconds to wait
+
+        """
+        timeout = timeout if timeout else 600
+        log.info(f"Waiting for the build to reach {status} state")
+        jenkins_builds = self.get_builds_obj()
+        for jenkins_build in jenkins_builds:
+            if jenkins_build.name not in self.build_completed:
```

Comment:
Any reason for using this condition, it will check for the ele in the list one by one. It proceeds to the next ele  only if the current one has reached to the expected resource state.
Please clarify

Reply:
I add all the builds that reach to 'complete' state to a list, so I don't have to check them again.